tests/testsuite/rustup.rs

```diff
@@ -10,8 +10,16 @@ use std::path::{Path, PathBuf};
 
 use crate::prelude::*;
 use crate::utils::cargo_process;
-use cargo_test_support::paths::{home, root};
-use cargo_test_support::{process, project, str};
+use cargo_test_support::install::assert_has_installed_exe;
+use cargo_test_support::paths::{cargo_home, home, root};
+use cargo_test_support::registry::Package;
+use cargo_test_support::{execs, process, project, str};
+
+fn pkg(name: &str, vers: &str) {
+    Package::new(name, vers)
+        .file("src/main.rs", "fn main() {{}}")
+        .publish();
+}
```

Comment:
imo this abstraction does not carry enough value to outweigh what it is hiding.  We should instead just have these calls in the tests

Reply:
Looks like I had same the same thing before (https://github.com/rust-lang/cargo/pull/16131#discussion_r2446151094) but it got resolved